src/mumble/AudioOutputSpeech.cpp

```diff
@@ -51,7 +51,7 @@ AudioOutputSpeech::AudioOutputSpeech(ClientUser *user, unsigned int freq, Messag
 	// opus supports frames with: 2.5, 5, 10, 20, 40 or 60 ms of audio data.
 	// sample rate / 100 means 10ms mono audio data per frame.
 	iFrameSizePerChannel = iFrameSize = iSampleRate / 100; // for mono stream
-	iAudioBufferSize = iFrameSize * 12; // used to store decoded pcm data. is it too big for 4-bytes float?
+	iAudioBufferSize = iFrameSize * 4; // used to store decoded pcm data. is it too big for 4-bytes float?
```

Comment:
What's the rational behind this change?
Also what does this number even mean? Is this the size of a float?

Reply:
Yes. float has a length of 4. I can't see any reason to make it 12.

Reply:
The factor was introduced in https://github.com/mumble-voip/mumble/commit/0a6aa0199391c5d01cc1a8d4725e01925ab02f2f

and the FIXME comment was introduced in https://github.com/mumble-voip/mumble/commit/7586a61226a5477ff1f6418f51acdab65fe093f4#diff-6342de3f25802c029b8d3f892cf0543cR397

But I still can't find where the number 12 is coming from :thinking: 

----

Some further investigation brought up this comment here:
https://github.com/mumble-voip/mumble/blob/6e12d3120e8d850531b5be8889cf7cf5fe4cdfb5/src/mumble/AudioOutputSpeech.cpp#L213-L215

It seems to have something to do with Opus and from that comment I take it that normal Opus uses something that's 12 bytes long :thinking:

I think it might be related to how `iFrameSize` is determined:
https://github.com/mumble-voip/mumble/blob/0a6aa0199391c5d01cc1a8d4725e01925ab02f2f/src/mumble/AudioOutputSpeech.cpp#L58-L60

How long (in bytes) is an Opus frame?

Reply:
Ohhhh! Thanks for looking into the history!

Actually I read the code again and now I kinda understand what this 12 means. I read a line in Opus' documentation:

> To encode a frame, opus_encode() or opus_encode_float() must be called with exactly one frame (2.5, 5, 10, 20, 40 or 60 ms) of audio data.

in L346, this 12 * 10 ms is used to handle the max frame size 60ms, *2 for stereo data.

Now everything makes total sense for me.

And now I understand how large the buffer should be initialized. It should be not only the maximum buffer size supported by the system, but also need to be added by the size of 60ms stereo audio data.
`iBufferSize = iOutputSize + (system audio buffer upper bound)`

I will fix things up and document my findings today.

Reply:
I still don't understand it xD
Maybe I'll just wait for your documentation and see if this clarifies things ^^

L346 in which file? (if you go to the file on GitHub you can copy a permalink and if you insert that in your commit, it'll actually show a preview of the code-snippet)